ui/packages/tidb-dashboard-lib/src/apps/Statement/pages/Detail/PlanBind.tsx

```diff
@@ -0,0 +1,303 @@
+  const { t } = useTranslation()
+  const { data: planBindingStatus } = useClientRequest((reqConfig) =>
+    ctx!.ds.statementsPlanBindStatusGet!(
+      query.digest!,
+      query.beginTime!,
+      query.endTime!,
+      reqConfig
+    )
+  )
+
+  const [hasPlanToBind, setHasPlanToBind] = useState(false)
+  const [boundPlanDigest, setBoundPlanDigest] = useState<string | null>(null)
+  const [showPlanBindModal, setShowPlanBindModal] = useState(false)
+
+  const handleModalVisibility = (visibility: boolean) => {
+    setShowPlanBindModal(visibility)
+  }
+
+  const handleSetBoundPlanDigets = (planDigest: string | null) => {
+    setBoundPlanDigest(planDigest)
+  }
+
+  useEffect(() => {
+    if (planBindingStatus) {
+      setBoundPlanDigest(planBindingStatus.plan_digest!)
+    }
+  }, [planBindingStatus])
+
+  useEffect(() => {
+    plans.forEach((plan) => {
+      if (plan.plan_digest) {
+        setHasPlanToBind(true)
+      }
+    })
+  }, [plans])
+
+  return (
+    <Space align="center">
+      {!hasPlanToBind ? (
+        <Tooltip
+          title={t('statement.pages.detail.plan_bind.bound_available_tooltip')}
+          placement="leftTop"
+        >
+          <InfoCircleOutlined /> Unavailable
+        </Tooltip>
+      ) : (
+        <>
+          {boundPlanDigest ? (
+            <Space>
+              <span className={styles.GreenDot} />
+              {t('statement.pages.detail.plan_bind.bound')}
+            </Space>
+          ) : (
+            <Space>
+              <span className={styles.GreyDot} />
+              {t('statement.pages.detail.plan_bind.not_bound')}
+            </Space>
+          )}
+        </>
+      )}
+
+      <Button
+        onClick={() => handleModalVisibility(true)}
+        disabled={!hasPlanToBind}
+      >
+        {t('statement.pages.detail.plan_bind.title')}
+      </Button>
+      <PlanBindModal
+        showPlanBindModal={showPlanBindModal}
+        boundPlanDigest={boundPlanDigest}
+        plans={plans}
+        sqlDigest={query.digest!}
+        onHandleModalVisibility={handleModalVisibility}
+        onHandleSetBoundPlanDigets={handleSetBoundPlanDigets}
+      />
+    </Space>
+  )
+}
+
+interface PlanBindModalProps {
+  showPlanBindModal: boolean
+  boundPlanDigest: string | null
+  plans: StatementModel[]
+  sqlDigest: string
+  onHandleModalVisibility: (visibility: boolean) => void
+  onHandleSetBoundPlanDigets: (planDigest: string | null) => void
+}
+
+const PlanBindModal = ({
+  showPlanBindModal,
+  boundPlanDigest,
+  plans,
+  sqlDigest,
+  onHandleModalVisibility,
+  onHandleSetBoundPlanDigets
+}: PlanBindModalProps) => {
+  const ctx = useContext(StatementContext)
+  const { t } = useTranslation()
+  const handleOnCancel = () => {
+    onHandleModalVisibility(false)
+  }
+  const [selectedPlan, setSelectedPlan] = useState<string | null>(null)
+  const [isBinding, setIsBinding] = useState(false)
+  const [isDropping, setIsDropping] = useState(false)
```

Comment:
how about replace isBinding and is Dropping by isLoading.

Reply:
+1